# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# Отображаем уникальные значения в столбце 'children'
data['children'].unique()

# Оставляем только строки с количеством детей отличным от -1 и 20
data = data[(data['children'] != -1) & (data['children'] != 20)]

# Определяем функцию для категоризации количества детей
def categorize_children(children):
    if children == 0:
        return 'Бездетные'
    elif 1 <= children <= 2:
        return 'Семьи с 1-2 детьми'
    elif children >= 3:
        return 'Многодетные семьи'

# Применяем функцию к столбцу с количеством детей и создаем новый столбец 'children_category'
data['children_category'] = data['children'].apply(categorize_children)

# Рассчитываем долю должников для каждой категории
debtor_ratios = (data[data['debt'] == 1]['children_category'].value_counts() / data['children_category'].value_counts()) * 100

# Создаем сводную таблицу для расчета доли должников по каждой категории
pivot_table = pd.pivot_table(data, values='debt', index='children_category', aggfunc='mean') * 100

# Переименовываем столбец для ясности
pivot_table.rename(columns={'debt': 'Доля задолженности'}, inplace=True)

# Рассчитываем количество должников для каждой категории
debtors_count = data['children_category'].value_counts()

# Создаем DataFrame с информацией о доле должников и количестве клиентов в каждой категории
children_category_debtor_ratio = pd.DataFrame({'Доля задолженности': debtor_ratios, 'Количество клиентов в категории': debtors_count})

# Переименовываем индекс в 'Количество детей'
children_category_debtor_ratio = children_category_debtor_ratio.rename_axis('Количество детей')

# Отображаем обновленную сводную таблицу
pd.options.display.float_format = '{:.2f}%'.format
display(children_category_debtor_ratio)

Доля задолженности  Количество клиентов в категории
Количество детей                                                       
Бездетные                        7.54%                            14091
Семьи с 1-2 детьми               9.30%                             6860
Многодетные семьи                8.16%                              380

**Вывод:** 
Исходя из предоставленной таблицы, можно сделать следующие выводы относительно зависимости между количеством детей и возвратом кредита в срок:

 - Бездетные: Доля задолженности составляет 7.51%, а количество клиентов в этой категории - 14149.
 - Семьи с 1-2 детьми: Доля задолженности равна 9.28%, а количество клиентов - 6873.
 - Многодетные семьи: Доля задолженности составляет 8.16%, а количество клиентов - 380.

_Из этих данных можно предположить, что клиенты с детьми (как в категории "Семьи с 1-2 детьми", так и в категории "Многодетные семьи") имеют немного большую долю задолженности по сравнению с бездетными._

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [23]:
# Рассчитываем долю должников для каждой категории
debtor_ratios = (data[data['debt'] == 1]['family_status'].value_counts() / data['family_status'].value_counts()) * 100

# Создаем сводную таблицу для расчета доли должников по каждой категории
pivot_table = pd.pivot_table(data, values='debt', index='family_status', aggfunc='mean') * 100

# Переименовываем столбец для ясности
pivot_table.rename(columns={'debt': 'Доля задолженности'}, inplace=True)

# Рассчитываем количество должников для каждой категории
debtors_count = data['family_status'].value_counts()

# Создаем DataFrame с информацией о доле должников и количестве клиентов в каждой категории
family_status_debtor_ratio = pd.DataFrame({'Доля задолженности': debtor_ratios, 'Количество клиентов в категории': debtors_count})

# Переименовываем индекс в 'Семейное положение'
family_status_debtor_ratio = family_status_debtor_ratio.rename_axis('Семейное положение')

# Отображаем обновленную сводную таблицу
pd.options.display.float_format = '{:.2f}%'.format
display(family_status_debtor_ratio)

Доля задолженности  Количество клиентов в категории
Семейное положение                                                        
женат / замужем                     7.56%                            12261
гражданский брак                    9.31%                             4134
Не женат / не замужем               9.76%                             2796
в разводе                           7.06%                             1189
вдовец / вдова                      6.62%                              951

**Вывод:** 
Из предоставленной таблицы можно сделать следующие выводы относительно зависимости между семейным положением и возвратом кредита в срок:

 - Женат/замужем: Доля задолженности составляет 7.54%, а количество клиентов в этой категории - 12302.
 - Гражданский брак: Доля задолженности равна 9.25%, а количество клиентов - 4160.
 - Не женат/не замужем: Доля задолженности составляет 9.75%, а количество клиентов - 2799.
 - В разводе: Доля задолженности равна 7.06%, а количество клиентов - 1189.
 - Вдовец/вдова: Доля задолженности составляет 6.62%, а количество клиентов - 952.

_Из этих данных можно предположить, что клиенты, находящиеся в браке (женат/замужем), а также вдовцы/вдовы, имеют более низкую долю задолженности по сравнению с теми, кто не состоит в браке (гражданский брак, не женат/не замужем)._

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [24]:
# Определяем функцию для категоризации уровня дохода
def categorize_income(income):
    if 0 <= income <= 30000:
        return 'E'
    elif 30001 <= income <= 50000:
        return 'D'
    elif 50001 <= income <= 200000:
        return 'C'
    elif 200001 <= income <= 1000000:
        return 'B'
    elif income >= 1000001:
        return 'A'

# Применяем функцию к столбцу с общим доходом и создаем новый столбец 'income_category'
data['income_category'] = data['total_income'].apply(categorize_income)

# Рассчитываем долю должников для каждой категории
debtor_ratios = (data[data['debt'] == 1]['income_category'].value_counts() / data['income_category'].value_counts()) * 100

# Создаем сводную таблицу для расчета доли должников по каждой категории
pivot_table = pd.pivot_table(data, values='debt', index='income_category', aggfunc='mean') * 100

# Переименовываем столбец для ясности
pivot_table.rename(columns={'debt': 'Доля задолженности'}, inplace=True)

# Рассчитываем количество должников для каждой категории
debtors_count = data['income_category'].value_counts()

# Создаем DataFrame с информацией о доле должников и количестве клиентов в каждой категории
income_category_debtor_ratio = pd.DataFrame({'Доля задолженности': debtor_ratios, 'Количество клиентов в категории': debtors_count})

# Переименовываем индекс в 'Уровень дохода'
income_category_debtor_ratio = income_category_debtor_ratio.rename_axis('Уровень дохода')

# Отображаем обновленную сводную таблицу
pd.options.display.float_format = '{:.2f}%'.format
display(income_category_debtor_ratio)

Доля задолженности  Количество клиентов в категории
Уровень дохода                                                     
A                            8.00%                               25
B                            7.06%                             5014
C                            8.50%                            15921
D                            6.02%                              349
E                            9.09%                               22

**Вывод:** 
Из представленной таблицы можно сделать следующие выводы относительно зависимости между уровнем дохода и возвратом кредита в срок:

 - Уровень дохода C: Доля задолженности составляет 8.55%, а количество клиентов в этой категории - 13831.
 - Уровень дохода B: Доля задолженности равна 7.06%, а количество клиентов - 5013.
 - Уровень дохода D: Доля задолженности составляет 6.02%, а количество клиентов - 349.
 - Уровень дохода A: Доля задолженности равна 8.00%, однако количество клиентов в этой категории невелико (25).
 - Уровень дохода E: Доля задолженности составляет 9.09%, также при небольшом количестве клиентов в этой категории (22).
 
_Эти данные могут указывать на то, что клиенты с более высоким уровнем дохода (категории B и C) имеют относительно низкую долю задолженности по сравнению с клиентами с более низким уровнем дохода (категории D и E). Однако необходимо учитывать, что размер выборки в некоторых категориях (например, уровень дохода A и E) может быть слишком мал, что делает выводы менее надежными._

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [25]:
# Определяем функцию для категоризации целей кредита
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'Операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'Операции с недвижимостью'
        elif 'свад' in row:
            return 'Проведение свадьбы'
        elif 'образов' in row:
            return 'Получение образования'
    except:
        return 'Нет категории'

# Применяем функцию к столбцу с целями кредита и создаем новый столбец 'purchase_category'
data['purchase_category'] = data['purpose'].apply(categorize_purpose)

# Рассчитываем долю должников для каждой категории
debtor_ratios = (data[data['debt'] == 1]['purchase_category'].value_counts() / data['purchase_category'].value_counts()) * 100

# Создаем сводную таблицу для расчета доли должников по каждой категории
pivot_table = pd.pivot_table(data, values='debt', index='purchase_category', aggfunc='mean') * 100

# Переименовываем столбец для ясности
pivot_table.rename(columns={'debt': 'Доля задолженности'}, inplace=True)

# Рассчитываем количество должников для каждой категории
debtors_count = data['purchase_category'].value_counts()

# Создаем DataFrame с информацией о доле должников и количестве клиентов в каждой категории
purchase_category_debtor_ratio = pd.DataFrame({'Доля задолженности': debtor_ratios, 'Количество клиентов в категории': debtors_count})

# Переименовываем индекс в 'Цель кредита'
purchase_category_debtor_ratio = purchase_category_debtor_ratio.rename_axis('Цель кредита')

# Отображаем обновленную сводную таблицу
pd.options.display.float_format = '{:.2f}%'.format
display(purchase_category_debtor_ratio)


Доля задолженности  Количество клиентов в категории
Цель кредита                                                                 
Операции с недвижимостью               7.26%                            10751
Операции с автомобилем                 9.35%                             4279
Получение образования                  9.25%                             3988
Проведение свадьбы                     7.91%                             2313

**Вывод:** 
Из представленной таблицы можно сделать следующие выводы относительно влияния разных целей кредита на его возврат в срок:

 - Операции с недвижимостью: Доля задолженности составляет 7.24%, а количество клиентов в этой категории - 10780.

 - Операции с автомобилем: Доля задолженности равна 9.33%, а количество клиентов - 4288. Это может свидетельствовать о том, что клиенты, берущие кредит на автомобиль, имеют более высокий риск задолженности.

 - Получение образования: Доля задолженности составляет 9.23%, а количество клиентов - 3997. Клиенты, берущие кредит на образование, также имеют высокий процент задолженности.

 - Проведение свадьбы: Доля задолженности равна 7.83%, а количество клиентов - 2337. Клиенты, берущие кредит на свадьбу, имеют более низкий процент задолженности по сравнению с кредитами на автомобиль и образование.

_Эти данные могут быть полезны для банковской индустрии при принятии решений о выдаче кредитов, учитывая, что разные цели кредита могут представлять разные уровни риска для возврата._

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* 

Ошибки в данных могут быть вызваны как технологическими факторами, так и человеческими ошибками.

__Технологические ошибки:__
Такие ошибки происходят при копировании, считывании, смене формата и могут возникнуть даже во время записи данных. Примером технологической ошибки может служить изменение формата данных или наличие символов, мешающих корректному считыванию. 

__Человеческие ошибки:__
Люди могут допускать ошибки при вводе данных, опечатывать, указывать неверный порядок величин, перепутывать числа или категориальные значения. Эти ошибки могут быть вызваны невнимательностью, недостатком знаний, трудностями в интерпретации инструкций и другими факторами. 

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 
Заполнение пропусков медианным значением является распространенным и предпочтительным подходом для обработки пропущенных данных в количественных переменных, так как в отличие от среднего значения (среднего арифметического), медиана менее чувствительна к большим выбросам. Это предотвращает искажение распределения в случае наличия аномальных значений.

### Шаг 4: общий вывод.

 - Количество детей:
Бездетные клиенты имеют более низкую долю задолженности (7.51%) по сравнению с семьями с 1-2 детьми (9.28%) и многодетными семьями (8.16%).

 - Семейное положение:
Вдовцы/вдовы и женатые/замужние клиенты имеют более низкие доли задолженности (6.62% и 7.54% соответственно), в то время как клиенты в гражданском браке и не состоящие в браке имеют более высокие доли задолженности (9.25% и 9.75%).

 - Уровень дохода:
Клиенты с уровнем дохода B имеют более низкую долю задолженности (7.06%), в то время как клиенты с уровнем дохода E имеют самую высокую долю задолженности (9.09%).

 - Цель кредита:
Клиенты, берущие кредит на образование (9.23%) или автомобиль (9.33%), имеют более высокие доли задолженности по сравнению с клиентами, берущими кредит на свадьбу (7.83%) или операции с недвижимостью (7.24%).

__Итог:__
_Семейное положение, количество детей, уровень дохода и цель кредита оказывают влияние на долю задолженности._